src/test/java/com/github/fge/jsonpatch/PathParserTest.java

```diff
@@ -0,0 +1,149 @@
+package com.github.fge.jsonpatch;
+
+import org.testng.annotations.Test;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertThrows;
+
+public class PathParserTest {
```

Comment:
There are no tests here for top-level fields (without parent). 
* 'id'
* '/id'
* '$.id'
* '$['id']